### ISSUE TITLE
very large vmem usage

### ISSUE THREAD

**Issue by bcdarwin** [Mouse-Imaging-Centre/pydpiper#378]
For better or worse, vmem actually matters due to scheduler configuration.

Seems to be unrelated to any Pydpiper code changes; associated with these sites:

```
# pipeline_executor.py(29)
from pyminc.volumes.volumes import mincException
# pipeline_executor.py(641)
from pydpiper.core.arguments import _mk_execution_parser
```

The first one appears to be related to the transitive import of numpy (see https://github.com/numpy/numpy/issues/10455) and can indeed be alleviated by setting OMP_NUM_THREADS (could be done internal to Pydpiper) or by installing numpy from source by passing `--no-binary :all:` to pip.  EDIT: The remainder of the memory is due to `import networkx` using >1GB of virtual memory.  This isn't needed on the executor side at all ...

TODO: add an --executor-memory-overhead flag or similar for constant memory offsets ...

**Comment by bcdarwin** [comment]
Half of this is fixed in 14a3276; the other half could be solved by setting OMP_NUM_THREADS before doing any imports.  This probably deserves a mention on the INSTALL page.

**Comment by bcdarwin** [comment]
A more ideal fix could involve the [threadpoolctl](https://github.com/joblib/threadpoolctl) library.
